src/parser.js

```diff
@@ -704,6 +704,23 @@ export default class Parser extends Stream {
             },
             'independent-segments'() {
               this.manifest.independentSegments = true;
+            },
+            'i-frame-playlist'() {
+              if (!this.manifest.iFramePlaylists) {
+                this.manifest.iFramePlaylists = [];
```

Comment:
Consider adding a default empty list in the manifest declaration instead of this if statement.

Reply:
Thanks, that's better!

Reply:
@dzianis-dashkevich  now I remember why I didn't add it to the manifest declaration, because it meant updating a ton of unit tests. If it's OK for you, I'll make the changes this weekend, as I've done enough unit testing for today. 😅

In the meantime, I think it would be best to remove your validation from this PR so that it doesn't get merged by mistake.

Reply:
@dzianis-dashkevich I did a force push to include the change and update the commit message. I also added two commits, one updating the test fixtures and the other containing the refactoring. Finally, I rebase main into this branch.



![harold](https://github.com/videojs/m3u8-parser/assets/34163393/b75ea884-deb3-4059-86a5-1a1be4f2158c)

Reply:
I feel your pain... Thank you so much for your outstanding contribution!